modules/internal/ChapelArray.chpl

```diff
@@ -2245,9 +2245,13 @@ module ChapelArray {
       _value.dsiSerialRead(f);
     }
 
+    proc IRV where !isSparseArr(this) {
```

Comment:
I'd suggest leaving off this where clause.  Since the two IRV functions cover all cases with where clauses, it leaves no room for other overloads, e.g. for a more specific error message.